test/integration/test_calls.py

```diff
@@ -0,0 +1,509 @@
+                    self.api_instance.update_call(BW_ACCOUNT_ID, callId, body, _return_http_data_only=False)
+                elif get_call_response[0].state == 'complete':
+                    callIdArray.remove(callId)
+            callIdArray.clear()
+        pass                
+
+    def assertApiException(self, context: ApiException, expectedException: ApiException, expected_status_code: int):
+        """Validates that common API exceptions, (401, 403, and 404) are properly formatted
+        Args:
+            context (ApiException): Exception to validate
+            expectedException (ApiException): Expected exception type
+            expected_status_code (int): Expected status code
+        """
+        self.assertIs(type(context.exception), expectedException)
+        self.assertIs(type(context.exception.status), int)
+        self.assertEqual(context.exception.status, expected_status_code)
+        self.assertIs(type(context.exception.body), str)
+
+    def test_create_call(self):
+        """
+            Validate a Create Call request with all optional parameters
+        """
+        time.sleep(3)        
+        answer_url = BASE_CALLBACK_URL
+        call_body = CreateCall(
+            to=USER_NUMBER, 
+            _from=BW_NUMBER, 
+            application_id=BW_VOICE_APPLICATION_ID, 
+            answer_url=answer_url,
+            answer_method=CallbackMethodEnum("POST"),
+            username="mySecretUsername",
+            password="mySecretPassword1!",
+            answer_fallback_url="https://www.myFallbackServer.com/webhooks/answer",
+            answer_fallback_method=CallbackMethodEnum("POST"),
+            fallback_username="mySecretUsername",
+            fallback_password="mySecretPassword1!",
+            disconnect_url="https://myServer.com/bandwidth/webhooks/disconnectUrl",
+            disconnect_method=CallbackMethodEnum("POST"),
+            call_timeout=30.0,
+            callback_timeout=15.0,
+            machine_detection=MachineDetectionConfiguration(
+                mode=MachineDetectionModeEnum("async"),
+                detection_timeout=15.0,
+                silence_timeout=10.0,
+                speech_threshold=10.0,
+                speech_end_threshold=5.0,
+                machine_speech_end_threshold=5.0,
+                delay_result=False,
+                callback_url="https://myServer.com/bandwidth/webhooks/machineDetectionComplete",
+                callback_method=CallbackMethodEnum("POST"),
+                username="mySecretUsername",
+                password="mySecretPassword1!",
+                fallback_url="https://myFallbackServer.com/bandwidth/webhooks/machineDetectionComplete",
+                fallback_method=CallbackMethodEnum("POST"),
+                fallback_username="mySecretUsername",
+                fallback_password="mySecretPassword1!",
+            ),
+            priority=5,
+            tag="tag_example",
+        )
+
```

Comment:
We could do that, but this particular call body is only used once for this initial test to make sure that we can send all of these parameters. The future call bodies that I use in the tests are smaller and I am making a separate global variable that gets reused for that.

Reply:
Agreed on it being not re-used. The suggestion is to reduce ~30 lines to a line(or two) so that it's easy to understand the test structure at high level. Doing it will help fellow developer to easily understand the test `setup` (create call), `action under test` and `validations`. To dig more into `create_call_setup` one simply click open the implementing method.